src/ir/item.rs

```diff
@@ -1194,7 +1194,18 @@ impl ClangItemParser for Item {
         // Types are sort of special, so to avoid parsing template classes
         // twice, handle them separately.
         {
-            let applicable_cursor = cursor.definition().unwrap_or(cursor);
+            let definition = cursor.definition();
+            let applicable_cursor = definition.unwrap_or(cursor);
+
+            if definition.is_some() && definition != Some(cursor) {
```

Comment:
Care to plop the commit message as a comment right here? I think that would help future readers of this code.